lib/mongoose_push/api/v3/response_encoder.ex

```diff
@@ -0,0 +1,44 @@
+defmodule MongoosePush.API.V3.ResponseEncoder do
+  @moduledoc """
+  Module for handling internal responses to V3 HTTP2 codes
+  """
+  @behaviour MongoosePush.API
+  alias MongoosePush.Service
+
+  @spec to_status(:ok | {:error, Service.error()} | {:error, MongoosePush.error()}) ::
+          {non_neg_integer, %{details: atom | String.t()} | nil}
+  def to_status(:ok), do: {200, nil}
+
+  def to_status({:error, {type, reason}}) when is_atom(reason) do
+    return_code =
+      case type do
+        :invalid_request -> 400
+        :unregistered -> 410
+        :payload_too_large -> 413
+        :too_many_requests -> 429
+        :auth -> 503
+        :service_internal -> 503
+        :internal_config -> 503
+        :unspecified -> 520
+        :generic -> 400
```

Comment:
That's not really true though. Only one specific value here should be 400 - `{:generic, :no_matching_pool}` (we can btw. send `reason => :no_matching_pool` in such case - `:generic` errors are controlled by MongoosePush, so we can match and actually use the `reason`).
All other values should be 500 for now. When we add proper handling for connection issues, we'll get also 503 from this.